flink-connector-akka/src/main/java/org/apache/flink/streaming/connectors/akka/AkkaSource.java

```diff
@@ -0,0 +1,147 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.akka;
+
+import akka.actor.ActorRef;
+import akka.actor.ActorSystem;
+import akka.actor.PoisonPill;
+import akka.actor.Props;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import org.apache.flink.api.common.functions.RuntimeContext;
+import org.apache.flink.api.common.functions.StoppableFunction;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.source.RichSourceFunction;
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+import org.apache.flink.streaming.api.operators.StreamingRuntimeContext;
+import org.apache.flink.streaming.connectors.akka.utils.ReceiverActor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of {@link SourceFunction} specialized to read messages
+ * from Akka actors.
+ */
+public class AkkaSource extends RichSourceFunction<Object>
+  implements StoppableFunction {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AkkaSource.class);
+
+  private static final long serialVersionUID = 1L;
+
+  // --- Fields set by the constructor
+
+  private final Class<?> classForActor;
+
+  private final String actorName;
+
+  private final String urlOfPublisher;
+
+  // --- Runtime fields
+  private transient ActorSystem receiverActorSystem;
+  private transient ActorRef receiverActor;
+  private transient Object waitLock;
+  private transient boolean running = true;
+
+  protected transient boolean autoAck;
+
+  /**
+   * Creates {@link AkkaSource} for Streaming
+   *
+   * @param actorName Receiver Actor name
+   * @param urlOfPublisher tcp url of the publisher or feeder actor
+   */
+  public AkkaSource(String actorName,
+          String urlOfPublisher) {
+    super();
+    this.classForActor = ReceiverActor.class;
+    this.actorName = actorName;
+    this.urlOfPublisher = urlOfPublisher;
+  }
+
+  @Override
+  public void open(Configuration parameters) throws Exception {
+    waitLock = new Object();
+    receiverActorSystem = createDefaultActorSystem();
+
+    RuntimeContext runtimeContext = getRuntimeContext();
+    if (runtimeContext instanceof StreamingRuntimeContext
+      && ((StreamingRuntimeContext) runtimeContext).isCheckpointingEnabled()) {
+      autoAck = false;
+    } else {
+      autoAck = true;
+    }
+  }
+
+  @Override
+  public void run(SourceFunction.SourceContext<Object> ctx) throws Exception {
+    LOG.info("Starting the Receiver actor {}", actorName);
+    receiverActor = receiverActorSystem.actorOf(
+      Props.create(classForActor, ctx, urlOfPublisher, autoAck), actorName);
+
+    running = true;
+    LOG.info("Started the Receiver actor {} successfully", actorName);
+
+    while (running) {
+      synchronized (waitLock) {
```

Comment:
I'm not an akka expert, but can't we call `receiverActorSystem.awaitTermination();` instead here?